Fix S_ISDOOR/S_IFDOOR spelling mistake.

This fix has been sent upstream.

--- tree-1.7.0/tree.c.orig	2016-03-03 06:04:25.447097615 -0800
+++ tree-1.7.0/tree.c	2016-03-03 06:04:49.600428587 -0800
@@ -1258,7 +1258,7 @@
   else if (m == S_IFIFO) return '|';
   else if (m == S_IFLNK) return '@'; /* Here, but never actually used though. */
 #ifdef S_IFDOOR
-  else if (m == S_ISDOOR) return '>';
+  else if (m == S_IFDOOR) return '>';
 #endif
   else if ((m == S_IFREG) && (mode & (S_IXUSR | S_IXGRP | S_IXOTH))) return '*';
   return 0;
